Add CallContext() to support calling with a context

iszak opened the pull request:
This allows to configure a timeout per call as opposed to a global
timeout that's currently supported. This is a **breaking** change so I'm happy to discuss whether we create a new method e.g. `CallContext` or do something similar like `http` where you can add a context to the request.

I believe this fixes https://github.com/hooklift/gowsdl/issues/111

iszak commented:
It appears this has a greater implication that it won't support Golang < 1.7 as this is when Context was introduced. We may be able to work around it, but it seems those versions are no longer supported, as per the [release policy](https://golang.org/doc/devel/release.html#policy) only go 1.11 and 1.10 are supported.

c4milo commented:
@iszak, we can also follow Go's stdlib convention of having `CallContext` type of functions to not break backwards compatibility.

iszak commented:
> @iszak, we can also follow Go's stdlib convention of having `CallContext` type of functions to not break backwards compatibility.
I believe I mentioned this is a possible approach in the original comment.

iszak commented:
@c4milo I have added `CallCtx` and reverted `Call`. Although, I would still consider this breaking backwards compatibility in terms of Golang versions - are we okay with that given it's EOL?

c4milo commented:
I'm not too concerned about backwards compatibility but we could tag a version before merging this PR to not upset people. @anjmao, thoughts?

c4milo commented on soap/soap.go:
```diff
@@ -228,8 +229,17 @@ func (s *Client) AddHeader(header interface{}) {
 }
 
+// CallCtx performs HTTP POST request with a context
+func (s *Client) CallCtx(ctx context.Context, soapAction string, request, response interface{}) error {
```
Let's keep it consistent with Go's standard library and name this `CallContext` instead.

c4milo commented:
LGTM @iszak, thank you! I only left one comment. I'll let @anjmao to make the final call.

c4milo commented:
@iszak once the conflict is sorted out, we can also merge this one. Thanks!

c4milo commented:
@iszak, this is looking pretty good, just let me know once you are ready to get it merged. Thanks again for all the contributions!

iszak commented:
@c4milo this is ready for re-review.
